### PR TITLE
Upgrade parity-wasm to v0.41

### PR DESCRIPTION
**Pull request by nhynes**
This gives us some helpful methods like [clear_custom_section](https://docs.rs/parity-wasm/0.41.0/parity_wasm/elements/struct.Module.html#method.clear_custom_section). Requires https://github.com/oasislabs/wasmi/pull/10

### CODE REVIEW

File: ethcore/wasm/Cargo.toml [hunk 1]
```diff
@@ -15,13 +15,13 @@ hmac-drbg = "=0.1.2"
...
-parity-wasm = "0.31"
-pwasm-utils = "0.2"
+parity-wasm = "0.41"
+pwasm-utils = "0.12"
 sha2 = "0.6"
 vm = { path = "../vm" }
 wasi-types = "0.1"
 wasm-macros = { path = "macros" }
-wasmi = { git = "https://github.com/oasislabs/wasmi" }
+wasmi = { git = "https://github.com/oasislabs/wasmi", branch = "parity-wasm-0.41" }
```
Comment (mitjat) [inline ethcore/wasm/Cargo.toml:24]: we can use master again now, right?

**Review by mitjat — CHANGES_REQUESTED** [review]
Thanks for figuring this out!

**Comment by mitjat** [comment]
(marking as "needs changes" just for visibility of that one comment; please go ahead and merge when oyu resolve that.)

File: ethcore/wasm/Cargo.toml (hunk 1 above)
Comment (nhynes) [reply ethcore/wasm/Cargo.toml:24]: I haven't merged it in. It'd be idea to have an end-to-end fix and also test the failing wasm to make sure that everything works before we make changes in 3 repos (that we might have to revert).

**Pull request closed without merging**